RealmSwift/Tests/ObjectCreationTests.swift

```diff
@@ -237,15 +237,74 @@ class ObjectCreationTests: TestCase {
     }
 
     func testUpdateWithNestedObjects() {
-    }
+        let standalone = SwiftPrimaryStringObject(object: ["primary", 11])
+        Realm().beginWrite()
+        let object = Realm().create(SwiftLinkToPrimaryStringObject.self, value: ["otherPrimary", standalone, [["primary", 12]]], update: true)
+        Realm().commitWrite()
+
+        let stringObjects = Realm().objects(SwiftPrimaryStringObject)
+        XCTAssertEqual(stringObjects.count, 1)
+        let persistedObject = object.object!
 
-    func testCreateWithObjectsFromAnotherRealm() {
+        XCTAssertEqual(persistedObject.intCol, 12)
+        XCTAssertNil(standalone.realm) // the standalone object should be copied, rather than added, to the realm
+        XCTAssertEqual(object.object!, persistedObject)
+        XCTAssertEqual(object.objects.first!, persistedObject)
     }
 
+    // This doesn't yet work, as RLMIsObjectValidForProperty doesn't take into account the target Realm,
+    // so we end up trying to just add a link to another Realm, which raises an exception.
+//    func testCreateWithObjectsFromAnotherRealm() {
+//        let values = [
+//            "boolCol": true as NSNumber,
+//            "intCol": 1 as NSNumber,
+//            "floatCol": 1.1 as NSNumber,
+//            "doubleCol": 11.1 as NSNumber,
+//            "stringCol": "b" as NSString,
+//            "binaryCol": "b".dataUsingEncoding(NSUTF8StringEncoding)! as NSData,
+//            "dateCol": NSDate(timeIntervalSince1970: 2) as NSDate,
+//            "objectCol": SwiftBoolObject(object: [true]) as AnyObject,
+//            "arrayCol": [SwiftBoolObject(object: [true]), SwiftBoolObject()]  as AnyObject,
+//        ]
+//
+//        realmWithTestPath().beginWrite()
+//        let otherRealmObject = realmWithTestPath().create(SwiftObject.self, value: values)
+//        realmWithTestPath().commitWrite()
+//
+//        Realm().beginWrite()
+//        let object = Realm().create(SwiftObject.self, value: otherRealmObject)
+//        Realm().commitWrite()
+//
+//        XCTAssertNotEqual(otherRealmObject, object)
+//        verifySwiftObjectWithDictionaryLiteral(object, dictionary: values, boolObjectValue: true, boolObjectListValues: [true, false])
+//    }
+
     func testUpdateWithObjectsFromAnotherRealm() {
+        realmWithTestPath().beginWrite()
+        let otherRealmObject = realmWithTestPath().create(SwiftLinkToPrimaryStringObject.self, value: ["primary", NSNull(), [["2", 2], ["4", 4]]])
+        realmWithTestPath().commitWrite()
+
+        Realm().beginWrite()
+        Realm().create(SwiftLinkToPrimaryStringObject.self, value: ["primary", ["10", 10], [["11", 11]]])
+        let object = Realm().create(SwiftLinkToPrimaryStringObject.self, value: otherRealmObject, update: true)
+        Realm().commitWrite()
+
+        XCTAssertNotEqual(otherRealmObject, object) // the object from the other realm should be copied into this realm
```

Comment:
please test count of objects in the target realm. Also test all object values are what they are expected to be.